nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmFacet.java

```diff
@@ -12,36 +12,63 @@
  */
 package org.sonatype.repository.helm;
 
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.Optional;
+
+import javax.annotation.Nullable;
+
+import org.sonatype.nexus.blobstore.api.Blob;
+import org.sonatype.nexus.common.collect.AttributesMap;
+import org.sonatype.nexus.common.hash.HashAlgorithm;
 import org.sonatype.nexus.repository.Facet;
 import org.sonatype.nexus.repository.storage.Asset;
 import org.sonatype.nexus.repository.storage.Bucket;
 import org.sonatype.nexus.repository.storage.Component;
 import org.sonatype.nexus.repository.storage.StorageTx;
+import org.sonatype.nexus.repository.view.Content;
+import org.sonatype.nexus.repository.view.Payload;
 import org.sonatype.repository.helm.internal.AssetKind;
 
+import com.google.common.base.Supplier;
+import com.google.common.collect.ImmutableList;
+
+import static org.sonatype.nexus.common.hash.HashAlgorithm.SHA1;
+import static org.sonatype.nexus.common.hash.HashAlgorithm.SHA256;
+
 /**
  * @since 1.0.next
  */
 @Facet.Exposed
 public interface HelmFacet
     extends Facet
 {
-  Asset findOrCreateAssetWithComponent(
-      final String assetPath,
-      final AssetKind assetKind,
-      final StorageTx tx,
-      final Bucket bucket,
-      final HelmAttributes chart);
+  List<HashAlgorithm> HASH_ALGORITHMS = ImmutableList.of(SHA1, SHA256);
```

Comment:
We don't need this in interface. It's a part of implementation. It's better to put it into `Util` class

Reply:
yeap, it is a good point